Adding troubleshooting topics (#24)
osisoft/PI-Adapter-DNP3-Docs: content/troubleshooting/troubleshoot-pi-adapter-for-dnp3.md

---
uid: TroubleshootPIAdapterForDNP3
---

# Troubleshoot PI Adapter for DNP3

PI Adapter for DNP3 provides troubleshooting features that enable you to verify adapter configuration, confirm connectivity, and view message logs. If you are unable to resolve issues with the adapter or need additional guidance, contact OSIsoft Technical Support through the [OSIsoft Customer Portal](https://my.osisoft.com/).

## Check configurations

Incorrect configurations can interrupt data flow and cause errors in values and ranges. Perform the following steps to confirm correct configuration for your adapter.

> 1. Navigate to [data source configuration](xref:PIAdapterForDNP3DataSourceConfiguration) and verify that <!-- Insert data source parameters that need to be checked --> are correct.

Author: ```suggestion
1. Navigate to [data source configuration](xref:PIAdapterForDNP3DataSourceConfiguration) and verify that <!-- Insert data source parameters that need to be checked  TEST--> are correct.
```

> 2. Navigate to [data selection configuration](xref:PIAdapterForDNP3DataSelectionConfiguration) and verify the following for each configured data selection item below:

Reviewer: This construction is awkward. 

```suggestion
2. Navigate to [data selection configuration](xref:PIAdapterForDNP3DataSelectionConfiguration) and verify that the following data selection items are correct:
```

    <!-- Insert data selection parameters that need to be checked-->

> 3. Navigate to [egress endpoints configuration](xref:EgressEndpointsConfiguration) and verify each configured endpoint's **Endpoint** property and credentials are correct.

Reviewer: This sentence is confusing because its construction isn't parallel. My suggestion:

```suggestion
3. Navigate to [egress endpoints configuration](xref:EgressEndpointsConfiguration). For each configured endpoint, verify that the **Endpoint** and authentication properties are correct.
```
